### PR TITLE
Create server in trial tests

### PR DESCRIPTION
**Pull request by cyli**
Add an API for creating servers directly with Nova for the LBSH tests.

Also change the `wait_for_servers` function to be able to match against all servers on the account, not just the autoscaled servers.  That way it can be used to wait for particular non-autoscale servers to become active.

### CODE REVIEW

**Comment by cyli** [comment]
jenkins please run integration tests

File: otter/integration/lib/nova.py [hunk 1]
```diff
@@ -120,35 +122,62 @@ def list_servers(rcs, pool, _treq=treq):
...
+
+    :return: the server ID of the created server.
+    """
+    d = _treq.post(
+        "{0}/servers".format(rcs.endpoints['nova']),
+        json.dumps(server_args),
+        headers=headers(str(rcs.token)),
+        pool=pool
+    ).addCallback(check_success, [202]).addCallback(_treq.json_content)
+    return d.addCallback(itemgetter('server')).addCallback(itemgetter('id'))
```
Comment (kivattik) [inline otter/integration/lib/nova.py]: I like this means of getting a value from the response. :)

File: otter/integration/lib/test_nova.py [hunk 2]
```diff
@@ -150,20 +150,33 @@ def setUp(self):
...
         d = nova.list_servers(self.rcs, pool=self.pool,
                               _treq=treq)
         self.assertEqual({'servers': {}}, self.successResultOf(d))
 
+    def test_create_server(self):
+        """
+        Create a server with particular args and succeeds on 202.
+        """
+        treq = get_fake_treq(self, 'post', 'novaurl/servers',
+                             (('{"server": {}}',), self.expected_kwargs),
```
Comment (kivattik) [inline otter/integration/lib/test_nova.py]: The args passed in is empty, but the created server has an id of 12345, which doesn't seem right.

File: otter/integration/lib/test_nova.py [hunk 3]
```diff
@@ -202,8 +215,8 @@ def test_wait_for_servers_retries_until_matcher_matches(self):
         """
         If the matcher does not matches the nova state, retries until it does.
```
Comment (kivattik) [inline otter/integration/lib/test_nova.py]: Already in existence and non-blocking, s/matches/match

Comment (cyli) [inline otter/integration/lib/test_nova.py]: Good catch, thanks!

**Comment by kivattik** [comment]
:+1:  With ninja edits :)

**Comment by cyli** [comment]
Thanks @kivattik!

**Pull request merged**